sdk/include/opentelemetry/sdk/logs/exporter.h

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <memory>
+#include <vector>
+#include "opentelemetry/logs/log_record.h"
+#include "opentelemetry/sdk/logs/processor.h"
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace sdk
+{
+namespace logs
+{
+/**
+ * ExportResult is returned as result of exporting a batch of Log Records.
+ */
+enum class ExportResult
+{
+  // The batch was exported successfully
+  kSuccess = 0,
+  // The batch was exported unsuccessfully and was dropped
+  kFailure
+};
+
+/**
+ * LogExporter defines the interface that log exporters must implement.
+ */
+class LogExporter
+{
+public:
+  virtual ~LogExporter() = default;
+
+  /**
+   * Exporters that implement this should typically format each LogRecord into the format
+   * required by the exporter destination (e.g. JSON), then send the LogRecord to the exporter.
+   * The exporter may retry logs a maximum of 3 times before dropping and returning kFailure.
+   * If this exporter is already shut down, should return kFailure.
+   * @param records: a vector of unique pointers to log records
+   * @returns an ExportResult code (whether export was success or failure)
+   *
+   * TODO: This method should not block indefinitely. Should abort within timeout.
```

Comment:
Can you comment on concurrency expectations of this?  Is it expected to block and only be called on a single thread?   Is it expected to be fully concurrent and thread-safe?

Reply:
Yes, it should only be called on a single thread, based on this [spec](https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/sdk.md#exportbatch) - "the `Export` method should only be called again after the current call returns". Aborting the method within a certain timeout should ensure it doesn't block indefinitely

Reply:
This [spec](https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/sdk.md#exportbatch) states that exporter should only be called on a single thread. , and the `Export` method should only be called again after the current call returns". So aborting the method within a certain timeout should ensure it doesn't block indefinitely. I don't think there should be any concurrency issues here.

Reply:
Right, I don't think you should make the users read the spec to know this class should only be used on a single thread.

I'm asking for you to add that to the documentation of the class so someone implementing it sees what they need to know to follow the spec.

Reply:
Ah gotcha, will add that in!